### ISSUE TITLE
Myriad of errors on connect, no logs?

### ISSUE THREAD

**Issue by Helveg** [miguelgrinberg/Flask-SocketIO#1207]
I have an Angular app that connects to a `flask_socketio` app, in local development everything is fine, but when I deploy it with `mod_wsgi` it's very fickle: Even when I comment out all the code in the `.on('connect')` and with `always_connect=True` at best 40% of the requests return 200, others return 400 or 401. With `Websocket connection to ... failed: Error during WebSocket handshake: Unexpected response code: 400` being the most common. Or the connection is established but within a few seconds 1-3 requests return `Bad Request`. In local development it doesn't seem to send this many  requests

Before I was able to get some error output in the logs, and the `echo` of `sqlalchemy` also ends up in the logs, but `flask_socketio` reports nothing.

* How can I log the errors to a file?
* Could this be because of Apache and mod_wsgi? Would it be better to run it with eventlet and proxy it?

**Comment by miguelgrinberg** [comment]
mod_wsgi/Apache is not a supported deployment platform. See the docs for the deployment options that work.

**Comment by Helveg** [comment]
I switched to a proxy approach and that works!

**Comment by Helveg** [comment]
I'm still getting the same kind of errors,

**Comment by Helveg** [comment]
I'm assuming this is because Apache times out the proxy connection. Do you think [mod_proxy_wstunnel](https://httpd.apache.org/docs/trunk/mod/mod_proxy_wstunnel.html) could work?

Does flask_socketio require to setup the WebSocket connection itself or could it work to proxy it from Apache straight to the `wss://` address that the WS calls go to now?

**Comment by miguelgrinberg** [comment]
As I said above, Apache is not a supported platform. I don't use it myself, so I don't really know the answer to your questions.

If you figure out how to use it and would like to contribute to the documentation feel free to submit a PR.

**Comment by Helveg** [comment]
Ok I will try to contribute if I figure it out, I'll close this issue again because I have confirmed it's because Apache's proxy directives abort the connection after a timeout. 

Here's what I've found to work for my application, in case it helps others setup flask_socketio with Apache:

1. Enable `mod_proxy`, `mod_proxy_http`, `mod_proxy_balancer` and `mod_lbmethod_byrequests`.
```
  sudo a2enmod proxy
  sudo a2enmod proxy_http
  sudo a2enmod proxy_balancer
  sudo a2enmod lbmetho_byrequests
```
2. Add a `ProxyPass` , high `ProxyTimeout` and `TimeOut` directive to your VirtualHost directive:

```
  ProxyPass /proxy_point http://localhost:3000 connectiontimeout=36000 timeout=36000
```

This configuration however doesn't work for me and the timeout errors still happen frequently, but since the connection is automatically reestablished immediatly afterwards I'm going to ignore the problem ❤️ 

The real solution would be to get `mod_proxy_wstunnel` working, but I couldn't figure out the configuration yet. The proposed conventional solutions like `sudo a2enmod proxy_wstunnel` and then in the config also doesn't work for me:

```
  ProxyPass /proxy_point ws://localhost:3000
```

Here are some docs for people with the same problem:
https://httpd.apache.org/docs/trunk/mod/mod_proxy_wstunnel.html
https://stackoverflow.com/questions/27526281/websockets-and-apache-proxy-how-to-configure-mod-proxy-wstunnel
